### PR TITLE
fixed https://github.com/NuGet/Home/issues/1245

### PR DESCRIPTION
**Pull request by zhili1208**
https://github.com/NuGet/Home/issues/1245
the bug is ScriptPackage does not contain PackageAssemblyReference.
the fix is bring PackageAssemblyReference back for install.ps
@yishaigalatzer @emgarten @deepakaravindr @feiling

### CODE REVIEW

File: src/PackageManagement.VisualStudio/PackageAssemblyReference.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        public PackageAssemblyReference(string path)
+        {
+            _path = path;
+        }
+
+        public string Name
+        {
+            get
+            {
+                return System.IO.Path.GetFileName(_path);
```
Comment (emgarten) [inline src/PackageManagement.VisualStudio/PackageAssemblyReference.cs]: Add a using and make this Path.GetFileName(_path);

**Comment by emgarten** [comment]
This should use PackageFolderReader.GetReferenceItems() and GetNearest on the Any framework.

**Comment by emgarten** [comment]
:watch:

File: src/PackageManagement.VisualStudio/ScriptPackage.cs [hunk 2]
```diff
@@ -25,5 +35,30 @@ public string Version
...
+            }
+        }
+
+        private IEnumerable<IPackageAssemblyReference> GetAssemblyReferencesCore()
+        {
+            var nupkg = new DirectoryInfo(_installPath).GetFiles("*.nupkg", SearchOption.TopDirectoryOnly).FirstOrDefault();
+            var reader = new PackageReader(nupkg.OpenRead());
+            var referenceItems = reader.GetReferenceItems();
+
+            var files = NuGetFrameworkUtility.GetNearest<FrameworkSpecificGroup>(referenceItems, NuGetFramework.AnyFramework).Items;
```
Comment (emgarten) [inline src/PackageManagement.VisualStudio/ScriptPackage.cs]: this could be null

File: src/PackageManagement.VisualStudio/ScriptPackage.cs [hunk 3]
```diff
@@ -25,5 +35,30 @@ public string Version
...
+                    _assemblyReferences = GetAssemblyReferencesCore().ToList();
+                }
+
+                return _assemblyReferences;
+            }
+        }
+
+        private IEnumerable<IPackageAssemblyReference> GetAssemblyReferencesCore()
+        {
+            var nupkg = new DirectoryInfo(_installPath).GetFiles("*.nupkg", SearchOption.TopDirectoryOnly).FirstOrDefault();
```
Comment (emgarten) [inline src/PackageManagement.VisualStudio/ScriptPackage.cs]: make sure the directory is there

**Comment by emgarten** [comment]
:watch:

File: src/PackageManagement.VisualStudio/ScriptPackage.cs (hunk 3 above)
Comment (yishaigalatzer) [inline src/PackageManagement.VisualStudio/ScriptPackage.cs]: Directory.EnumerateFiles ...

File: src/PackageManagement.VisualStudio/ScriptPackage.cs [hunk 4]
```diff
@@ -25,5 +35,37 @@ public string Version
...
+            var result = new List<PackageAssemblyReference>();
+            if (Directory.Exists(_installPath))
+            {
+                var nupkg = new DirectoryInfo(_installPath).EnumerateFiles("*.nupkg").FirstOrDefault();
+                if (nupkg != null)
+                {
+                    var reader = new PackageReader(nupkg.OpenRead());
+                    var referenceItems = reader.GetReferenceItems();
+                    var files = NuGetFrameworkUtility.GetNearest<FrameworkSpecificGroup>(referenceItems,
+                                                                                         NuGetFramework.AnyFramework).Items;
```
Comment (emgarten) [inline src/PackageManagement.VisualStudio/ScriptPackage.cs]: GetNearest can return null, check it before retrieving Items

**Comment by emgarten** [comment]
:shipit:

**Pull request closed without merging**